### PR TITLE
Issue 98 and 85

### PR DESCRIPTION
**Pull request by Rita-C**

### CODE REVIEW

File: grails-app/conf/application.yml [hunk 1]
```diff
@@ -123,6 +123,15 @@ security:
         realm: ALA User Details
         path:
           - /status/**
+#ala:
```
Comment (sbearcsiro) [inline grails-app/conf/application.yml:126]: Shouldn't these be left un-commented for default values?

File: grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy [hunk 2]
```diff
@@ -165,6 +177,10 @@ class RegistrationController {
         }
     }
 
+    def quoteValue(String value) {
```
Comment (sbearcsiro) [inline grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy:180]: `quoteValue` should be optional switched via configuration like in ALA CAS
There should also be a configurable `encode` on the cookie value as well (defaults to false as ALA CAS does)

File: grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy [hunk 3]
```diff
@@ -150,8 +150,20 @@ class RegistrationController {
...
+                    render(view: "accountError", model: [msg: "Failed to update user profile - A user is already registered with the email address"])
+                    return
+                }
+                // and username and email address must be kept in sync
                 params.userName = params.email
+                // update cookie when email address changed https://github.com/AtlasOfLivingAustralia/userdetails/issues/98
+                // cookie config values need to be consistent with values in CAS
+                def cookieName = grailsApplication.config.ala.cookie.name
+                if (cookieName && cookieService.findCookie(cookieName)) {
+                    cookieService.setCookie(cookieName, quoteValue(params.email), grailsApplication.config.ala.cookie.maxAge, grailsApplication.config.ala.cookie.path,
```
Comment (sbearcsiro) [inline grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy:164]: Minor nitpick but I dislike the grailsApplication.config.foo.bar style, I prefer either injecting with `@Value` or using `grailsApplication.config.getProperty('foo.bar', Type, optionalDefault)`

File: grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy [hunk 4]
```diff
@@ -150,8 +150,20 @@ class RegistrationController {
...
             if (params.email != user.email) {
-                // email address has changed, and username and email address must be kept in sync
+                // email address has changed
+                if (userService.isEmailInUse(params.email, user)) {
+                    render(view: "accountError", model: [msg: "Failed to update user profile - A user is already registered with the email address"])
+                    return
+                }
+                // and username and email address must be kept in sync
                 params.userName = params.email
+                // update cookie when email address changed https://github.com/AtlasOfLivingAustralia/userdetails/issues/98
```
Comment (sbearcsiro) [inline grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy:160]: The cookie should only be updated if `userService.update` succeeds (ie need to move this into the success block below).

**Review by sbearcsiro — CHANGES_REQUESTED** [review]
Looks good, just some minor clean up needs to be done IMHO

File: grails-app/conf/application.yml (hunk 1 above)
Comment (Rita-C) [reply grails-app/conf/application.yml]: updated

File: grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy (hunk 3 above)
Comment (Rita-C) [reply grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy]: updated

File: grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy (hunk 4 above)
Comment (Rita-C) [reply grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy]: done. 
added unit tests.

**Review by sbearcsiro — APPROVED** [review]
LGTM

**Pull request merged**